src/views/Scan.js

```diff
@@ -1,26 +1,93 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
-import MenuButton from '../components/MenuButton';
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ *
+ * @format
+ * @flow
+ */
 
-const ScanScreen = props => {
-    return (
-        <View style={styles.container}>
-            <MenuButton navigation={props.navigation} />
-            <Text style={styles.text}>Scan</Text>
-        </View>
-    );
-}
+import React, {useState, useEffect} from 'react';
+import useNFC from '../utils/nfc';
+import {StyleSheet} from 'react-native';
+import {Container, Content, Button, Text, H3, H1} from 'native-base';
+import MenuButton from '../components/MenuButton';
 
 const styles = StyleSheet.create({
-    container: {
-        flex: 1,
-        backgroundColor: '#fff',
-        alignItems: 'center',
-        justifyContent: 'center',
-    },
-    text: {
-        fontSize: 30,
-    }
+  wrapper: {
+    backgroundColor: '#343338',
+    padding: 60,
+  },
+  content: {
+    margin: 20,
+    flex: 1,
+  },
+  header: {
+    flex: 1,
+  },
+  body: {
+    flex: 1,
+    justifyContent: 'flex-end',
+  },
+  button: {
+    marginVertical: 10,
+    justifyContent: 'center',
+    backgroundColor: '#18CDCD',
+  },
+  text: {
+    color: '#ffffff',
+    textAlign: 'center',
+  },
+  image: {
+    width: 30,
+    height: 30,
+    margin: 10,
+    marginLeft: 0,
+    marginRight: 0,
+  },
 });
 
-export default ScanScreen;
\ No newline at end of file
+const Scan = props => {
+  const redirect = () => {
+    setTimeout(async function() {
+      props.navigation.navigate('Test', {uid});
```

Comment:
I think its best to see if theres a user in the scan page, since there will be two different pages and we have to figure out which to redirect to right away.